ANT+_Android_SDK/Sample/ANT+ Plugin Sampler/source/src/main/java/com/dsi/ant/antplus/pluginsampler/pulsezones/Activity_PulseZonesFitness.java

```diff
@@ -151,11 +156,23 @@ public void subscribeToHrEvents() {
             }
 
             runOnUiThread(() -> {
+                //Synchronization of starting time of the readings and chronometer start
+                if(!chronometer.isRunning()) {
+                    startTimeInMillisec = System.currentTimeMillis();
+                    chronometer.start();
+                }
+
                 int realTime = (int)(System.currentTimeMillis() - startTimeInMillisec)/1000;
                 series.appendData(new DataPoint(realTime, computedHeartRate), true, 1000);
 
+                //Insert values into database
+                ContentValues values = new ContentValues();
```

Comment:
I would recommend moving all DB related operations in a separate class, for instance, called HRRecordsRepository, which have the CRUD operations implemented and as I remember well, you may need extra methods for statistics.

Reply:
done